naga/src/valid/handles.rs

```diff
@@ -228,56 +309,69 @@ impl super::Validator {
             | crate::TypeInner::Image { .. }
             | crate::TypeInner::Sampler { .. }
             | crate::TypeInner::AccelerationStructure
-            | crate::TypeInner::RayQuery => (),
+            | crate::TypeInner::RayQuery => None,
             crate::TypeInner::Pointer { base, space: _ } => {
                 handle.check_dep(base)?;
+                None
             }
-            crate::TypeInner::Array { base, .. } | crate::TypeInner::BindingArray { base, .. } => {
+            crate::TypeInner::Array { base, size, .. }
+            | crate::TypeInner::BindingArray { base, size, .. } => {
                 handle.check_dep(base)?;
+                match size {
+                    crate::ArraySize::Pending(pending) => match pending {
+                        crate::PendingArraySize::Expression(expr) => Some(expr),
+                        crate::PendingArraySize::Override(_) => None,
```

Comment:
Shouldn't check the override initializer as well?

Reply:
Hmm, yes, we should.

Reply:
Filed as #6880, fixed in #6882.